src/main/ank/2.3-free-monoids.md

```diff
@@ -0,0 +1,123 @@
+```Haskell
+class Monoid m where
+  mempty :: m
+  mappend :: m -> m -> m
+```
+```kotlin:ank:silent
+package arrow.typeclasses
+
+class ForMonoid private constructor() { companion object }
+typealias MonoidOf<A> = arrow.Kind<ForMonoid, A>
+fun <A> MonoidOf<A>.fix(): Monoid<A> = this as Monoid<A>
+
+interface Monoid<A> : Semigroup<A>, MonoidOf<A> {
+  fun empty(): A
+
+  fun Collection<A>.combineAll(): A =
+    if (isEmpty()) empty() else reduce { a, b -> a.combine(b) }
+
+  fun combineAll(elems: List<A>): A = elems.combineAll()
+  companion object
+}
+```
+................
+```Haskell
+instance Monoid [a] where
+  mempty = []
+  mappend = (++)
+```
+```kotlin
+package free
+
+import arrow.higherkind
+import arrow.typeclasses.Monoid
+
+operator fun <A>FreeMonoid<A>.plus(b: FreeMonoid<A>) = FreeMonoid.just(this.terms + b.terms)
+
+@higherkind
+sealed class FreeMonoid<S>(val terms: List<S>): FreeMonoidOf<S> {
+
+    internal data class Pure<A>(val a: List<A>) : FreeMonoid<A>(a)
```

Comment:
Any reason for this to be an ADT? Can't we do `class FreeMonoid<S>(val terms: List<S>): FreeMonoidOf<S>` instead?

Reply:
I suggest you check both Links to integrate the free monoid better with Arrow. 
- https://medium.com/@olxc/free-monads-explained-pt-1-a5c45fbdac30
- https://typelevel.org/cats/datatypes/freemonad.html

Feel free to ask questions.